src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,80 @@
+package duke;
+
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+public class Parser {
+	/**
+	 * Creates task and stores it in Task arraylist.
+	 *
+	 * @param tasks List of tasks.
+	 * @param taskType Type of task.
+	 * @param descriptions The descriptions of the task.
+	 * @return Message containing task result.
+	 */
+	protected static String parseInput(TaskList tasks, String taskType, String[] descriptions) {
+		String output = "";
+
+		switch (taskType) {
+		case "list":
+			output = tasks.listTasks();
+			break;
+		case "mark":
+			output = tasks.markTask(tasks.getTask(Integer.parseInt(descriptions[1]) - 1));
+			break;
+		case "unmark":
+			output = tasks.unmarkTask(tasks.getTask(Integer.parseInt(descriptions[1]) - 1));
+			break;
+		case "todo":
+			try {
+				if (descriptions.length != 2) {
+					throw new DukeException("OOPS!!! The description of a todo cannot be empty.\n");
+				}
+
+				output = tasks.addTask(new ToDo(descriptions[1]));
+			} catch (DukeException error) {
+				output = Ui.errorMsg(error.getMessage());
+			}
```

Comment:
Rather than have multiple `try-catch` blocks, maybe you can try

```java
try {
    switch (taskType) {
    case "list":
        // ...
        break;
    case "mark":
        // ...
        break;
    case "unmark":
        // ...
        break;

    /* remaining cases here */

    }
} catch (DukeException error) {
    output = Ui.errorMsg(error.getMessage());
}
```

to reduce code duplication.